src/dstack/_internal/server/services/gateways/__init__.py

```diff
@@ -54,6 +54,8 @@
 
 GATEWAY_CONNECT_ATTEMPTS = 9
 GATEWAY_CONNECT_DELAY = 10
+GATEWAY_CONFIGURE_ATTEMPTS = 40
```

Comment:
Any reasons to have this many attempts?

Reply:
In my testing, there were about 50 seconds from the moment the SSH tunnel was up to the moment the Gateway's HTTP server was ready to handle requests. I decided to add some margin, so here the max total delay is about 120 seconds (40 attempts with 3-second delays). We could use fewer attempts with longer delays, but then the user would have to wait longer.

I don't like 3-second delays either. Ideally, the Gateway could notify the Server when it's ready so that the Server could start interacting immediately. But that seems less trivial, so I didn't choose that path

Reply:
> the Gateway could notify the Server when it's ready so that the Server could start interacting immediately

Yeah, that's an option, but you're right – definitely out of scope of this PR.